native/screens/SignIn.jsx

```diff
@@ -1,103 +1,166 @@
-// import { useEffect, useState } from "react";
-// import { useNavigate } from "react-router-dom";
-// // assets
-// import { useDispatch, useSelector } from "react-redux";
-// import { loginUser } from "../../services/redux/slices/uiActions.js";
-// import "../SignUp/SignUp.scss";
-// import { handleChange } from "../../services/utils/formHandlers";
-// import { SingleActionButton } from "../components/Button/SingleActionButton";
-// import { checkEmailAuth, signOut, verify } from "../../services/api/users";
-import { StyleSheet, Text, View } from "react-native";
+import { useState, createRef, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { signOut, checkEmailAuth, verify } from '../services/api/users.js';
+import { loginUser } from '../services/redux/actions/uiActions.js';
+
+// Native Components
+import { 
+  StyleSheet, 
+  Text, 
+  View, 
+  TextInput, 
+  TouchableOpacity, 
+  Button,
+  Alert,
+} from "react-native";
 
-export const SignIn = () => {
-  return (
-    <View>
-      <Text>SignIn</Text>
-    </View>
-  );
-};
 
-// export const ReactSignIn = () => {
-//   const navigate = useNavigate();
-//   const dispatch = useDispatch();
-//   const { _id: userId } = useSelector((state) => state.ui.user);
-//   const [authError, setAuthError] = useState(null);
-//   const [noAccountError, setNoAccountError] = useState(null);
-//   const [loginInfo, setLoginInfo] = useState({
-//     email: "test@test.com",
-//     password: "test",
-//   });
+export const SignIn = ({navigation}) => {
+  // redux
+  const dispatch = useDispatch();
+  // error states
+  const [noAccountError, setNoAccountError] = useState(null);
+  const [authError, setAuthError] = useState(null);
+  // log in state
+  const [logInInfo, setLogInInfo] = useState({
+    email: '',
+    password: '',
+  });
+  // input references
+  const { emailInputRef, passwordInputRef } = createRef();
 
-//   useEffect(() => {
-//     signOut();
-//   }, []);
+  // sign out user on sign in screen render
+  useEffect(() => {
```

Comment:
not necessary, ideally a logged-in user should never see the SignIn screen again unless they log themselves out